File: 1370-count-number-of-nice-subarrays/count-number-of-nice-subarrays.cs
public class Solution {
    public int NumberOfSubarrays(int[] nums, int k) {
        int output = 0, sum = 0;
        Dictionary<int, int> prefixSum = new Dictionary<int, int>();
        prefixSum[sum] = 1;

        for (int i = 0; i < nums.Length; i++){
            sum += nums[i] % 2;
            if (prefixSum.ContainsKey(sum - k)){
                output += prefixSum[sum - k];
            }
            if (prefixSum.ContainsKey(sum)){
                prefixSum[sum]++;
            }
            else{
                prefixSum[sum] = 1;
            }
        }
        return output;
    }
}

File: 1370-count-number-of-nice-subarrays/count-number-of-nice-subarrays.java
class Solution {
    public int numberOfSubarrays(int[] nums, int k) {
        int output = 0, sum = 0;
        Map<Integer, Integer> prefixSum = new HashMap<>();
        prefixSum.put(sum, 1);

        for(int i = 0; i < nums.length; i++){
            sum += nums[i] % 2;
            if(prefixSum.containsKey(sum - k)){
                output += prefixSum.get(sum - k);
            }
            prefixSum.put(sum, prefixSum.getOrDefault(sum, 0) + 1);
        }
        return output;
    }
}